src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -6600,4 +6600,43 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="ERR_FunctionPointerTypesInAttributeNotSupported" xml:space="preserve">
     <value>Using a function pointer type in a 'typeof' in an attribute is not supported.</value>
   </data>
-</root>
+  <data name="ERR_BadCallerArgumentExpressionParamWithoutDefaultValue" xml:space="preserve">
+    <value>The CallerArgumentExpressionAttribute may only be applied to parameters with default values</value>
+  </data>
+  <data name="ERR_NoConversionForCallerArgumentExpressionParam" xml:space="preserve">
+    <value>CallerArgumentExpressionAttribute cannot be applied because there are no standard conversions from type '{0}' to type '{1}'</value>
+  </data>
+  <data name="WRN_CallerArgumentExpressionParamForUnconsumedLocation" xml:space="preserve">
+    <value>The CallerArgumentExpressionAttribute applied to parameter '{0}' will have no effect because it applies to a member that is used in contexts that do not allow optional arguments</value>
+  </data>
+  <data name="WRN_CallerArgumentExpressionParamForUnconsumedLocation_Title" xml:space="preserve">
+    <value>The CallerArgumentExpressionAttribute will have no effect because it applies to a member that is used in contexts that do not allow optional arguments</value>
+  </data>
+  <data name="WRN_CallerFilePathPreferredOverCallerArgumentExpression" xml:space="preserve">
+    <value>The CallerArgumentExpressionAttribute applied to parameter '{0}' will have no effect. It is overridden by the CallerFilePathAttribute.</value>
+  </data>
+  <data name="WRN_CallerFilePathPreferredOverCallerArgumentExpression_Title" xml:space="preserve">
+    <value>The CallerArgumentExpressionAttribute will have no effect; it is overridden by the CallerFilePathAttribute</value>
+  </data>
+  <data name="WRN_CallerLineNumberPreferredOverCallerArgumentExpression" xml:space="preserve">
+    <value>The CallerArgumentExpressionAttribute applied to parameter '{0}' will have no effect. It is overridden by the CallerLineNumberAttribute.</value>
+  </data>
+  <data name="WRN_CallerLineNumberPreferredOverCallerArgumentExpression_Title" xml:space="preserve">
+    <value>The CallerArgumentExpressionAttribute will have no effect; it is overridden by the CallerLineNumberAttribute</value>
+  </data>
+  <data name="WRN_CallerMemberNamePreferredOverCallerArgumentExpression" xml:space="preserve">
+    <value>The CallerArgumentExpressionAttribute applied to parameter '{0}' will have no effect. It is overriden by the CallerMemberNameAttribute.</value>
```

Comment:
>the [](start = 116, length = 3)

English nit: remove `the`

Reply:
Same for the other preferred warnings above.

---
In reply to: [602453392](https://github.com/dotnet/roslyn/pull/51952#discussion_r602453392) [](ancestors = 602453392)

Reply:
Should I do the same with existing warnings for consistency? e.g:

https://github.com/dotnet/roslyn/blob/64d1beefeca4a0f869cc8fb14fc903af676dcbe4/src/Compilers/CSharp/Portable/CSharpResources.resx#L4206

or only remove "The" from the new warnings?

Reply:
Of course it is... I guess just leave it as is.

Reply:
What we should do, though, is add a `remarks` section to indicate that `CallerArgumentExpressionAttribute` and `CallerMemberNameAttribute` should not be localized.

Reply:
@333fred There was a previous discussion somewhere with @AlekseyTs not to do this. I don't remember which PR though. Will try to find it.

Reply:
https://github.com/dotnet/roslyn/pull/51426#discussion_r581753713

Reply:
The localization team has asked us for these types notes before. I always support making sure we don't get future bugs on things that were translated and shouldn't have been.

Reply:
> What we should do, though, is add a remarks section to indicate that CallerArgumentExpressionAttribute and CallerMemberNameAttribute should not be localized.

> The localization team has asked us for these types notes before. I always support making sure we don't get future bugs on things that were translated and shouldn't have been.

FWIW, I am not aware of a requirement to add comments like that. Obviously, a developer can add remarks if he or she would like to do so. In this particular case, however, I believe that the suggested remark would state the obvious - type names are not translated and I personally wouldn't add it.    

---
In reply to: [605055933](https://github.com/dotnet/roslyn/pull/51952#discussion_r605055933) [](ancestors = 605055933)

Reply:
I think there was some issues with translation for things that looks "trivial" that shouldn't be translated.

See also https://github.com/dotnet/roslyn/issues/48364

Reply:
Also an example where "unmanaged" keyword was incorrectly translated, despite it's surrounded with single quotes which is a high indication it shouldn't be localized.

https://github.com/dotnet/roslyn/blob/02428e32a5e34dbaef1c27be3851ae8256a43841/src/Compilers/CSharp/Portable/xlf/CSharpResources.ja.xlf#L170-L174

Reply:
> I think there was some issues with translation for things that looks "trivial" that shouldn't be translated.

This is your PR feel free to add remarks that **you** feel appropriate.

---
In reply to: [605664888](https://github.com/dotnet/roslyn/pull/51952#discussion_r605664888) [](ancestors = 605664888)

Reply:
Well, let's check with the loc team.

@cristianosuzuki77 Is a "Locked" comment required for `CallerMemberNameAttribute` in a message like the following?

```xml
  <trans-unit id="WRN_CallerMemberNamePreferredOverCallerArgumentExpression">
    <source>The CallerArgumentExpressionAttribute applied to parameter '{0}' will have no effect. It is overriden by the CallerMemberNameAttribute.</source>
    <note />
  </trans-unit>
```